File: blog-admin/src/main/kotlin/com/blog/controller/upload/UploadController.java
package com.blog.controller.upload;

import com.blog.ApiResponse;
import com.blog.service.upload.UploadService;
import io.swagger.annotations.ApiOperation;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RestController
@RequiredArgsConstructor
public class UploadController {

    private final UploadService uploadService;

    @ApiOperation(value = "업로드", notes = "이미지 파일 업로드")
    @PostMapping("api/v1/admin/image/upload")
    public ApiResponse<String> imageUpload(@RequestPart MultipartFile file) {
        return ApiResponse.Companion.success(uploadService.imageUpload(file));
    }

}

File: blog-admin/src/main/kotlin/com/blog/controller/upload/UploadController.kt
package com.blog.controller.upload

import com.blog.ApiResponse
import com.blog.dto.upload.UploadRequest
import com.blog.service.upload.UploadService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestPart
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.multipart.MultipartFile

@RestController
class UploadController(
    private val uploadService: UploadService
) {
    @PostMapping("/image/upload")
    fun imageUpload(@RequestPart upload: MultipartFile, request: UploadRequest): ApiResponse<String?> {
        return ApiResponse.success(uploadService.imageUpload(upload, request.imageType))
    }
}